src/librustc_privacy/lib.rs

```diff
@@ -227,6 +242,9 @@ impl<'a, 'tcx> Visitor<'tcx> for EmbargoVisitor<'a, 'tcx> {
             hir::ItemKind::ExternCrate(..) => {}
         }
 
+        let orig_level = self.prev_level;
+        self.prev_level = item_level;
```

Comment:
Nit: `let orig_level = mem::replace(&mut self.prev_level, item_level);`

Reply:
Hmm, I'm not sure why these lines were moved from there to here actually.

Reply:
Ok. This was moved while I was figuring out how best to propagate the level from the `ReachEverythingInTheInterfaceVisitor`. Will move back and modify.

Reply:
Actually, this was moved so that `EmbargoVisitor::reach` would be able to see whether the parent node was `ReachableFromImplTrait` or at least `Reachable`, to propagate the correct level.